modulemd/meson.build

```diff
@@ -34,244 +36,413 @@ valgrind = find_program('valgrind', required: developer_build)
-    'v2/include/private/modulemd-dependencies-private.h',
-    'v2/include/private/modulemd-profile-private.h',
-    'v2/include/private/modulemd-defaults-private.h',
-    'v2/include/private/modulemd-defaults-v1-private.h',
-    'v2/include/private/modulemd-module-private.h',
-    'v2/include/private/modulemd-module-index-private.h',
-    'v2/include/private/modulemd-module-stream-private.h',
-    'v2/include/private/modulemd-module-stream-v1-private.h',
-    'v2/include/private/modulemd-module-stream-v2-private.h',
-    'v2/include/private/modulemd-service-level-private.h',
-    'v2/include/private/modulemd-subdocument-info-private.h',
-    'v2/include/private/modulemd-translation-private.h',
-    'v2/include/private/modulemd-translation-entry-private.h',
-    'v2/include/private/modulemd-util.h',
-    'v2/include/private/modulemd-yaml.h',
-    'v2/include/private/gi-binding-renames.h',
+modulemd_validator_v1 = executable(
+    'modulemd-validator-v1',
+    'modulemd-validator.c',
+    dependencies : [
+        modulemd_dep,
+    ],
+    install : true,
 )
 
-test_v2_srcs = files(
-    'v2/tests/test-modulemd-buildopts.c',
-    'v2/tests/test-modulemd-component-module.c',
-    'v2/tests/test-modulemd-component-rpm.c',
-    'v2/tests/test-modulemd-defaults.c',
-    'v2/tests/test-modulemd-defaults-v1.c',
-    'v2/tests/test-modulemd-dependencies.c',
-    'v2/tests/test-modulemd-merger.c',
-    'v2/tests/test-modulemd-module.c',
-    'v2/tests/test-modulemd-moduleindex.c',
-    'v2/tests/test-modulemd-modulestream.c',
-    'v2/tests/test-modulemd-profile.c',
-    'v2/tests/test-modulemd-rpmmap.c',
-    'v2/tests/test-modulemd-service-level.c',
-    'v2/tests/test-modulemd-translation.c',
-    'v2/tests/test-modulemd-translation-entry.c',
-    'v2/tests/test-utils.c',
-)
+header_path = 'modulemd'
 
-test_v2_priv_hdrs = files(
-    'v2/include/private/test-utils.h',
+install_headers(
+  modulemd_hdrs,
+  subdir : header_path,
 )
 
-test_v2_python_scripts = files(
-    'v2/tests/ModulemdTests/base.py',
-    'v2/tests/ModulemdTests/buildopts.py',
-    'v2/tests/ModulemdTests/componentrpm.py',
-    'v2/tests/ModulemdTests/defaults.py',
-    'v2/tests/ModulemdTests/defaultsv1.py',
-    'v2/tests/ModulemdTests/dependencies.py',
-    'v2/tests/ModulemdTests/merger.py',
-    'v2/tests/ModulemdTests/module.py',
-    'v2/tests/ModulemdTests/moduleindex.py',
-    'v2/tests/ModulemdTests/modulestream.py',
-    'v2/tests/ModulemdTests/profile.py',
-    'v2/tests/ModulemdTests/rpmmap.py',
-    'v2/tests/ModulemdTests/servicelevel.py',
-    'v2/tests/ModulemdTests/translation.py',
-    'v2/tests/ModulemdTests/translationentry.py',
+
+# Test env with release values
+test_release_env = environment()
+test_release_env.set('G_MESSAGES_DEBUG', 'all')
+test_release_env.set ('MESON_SOURCE_ROOT', meson.source_root())
+
+
+# Test env with fatal warnings and criticals
+test_env = test_release_env
+test_env.set('G_DEBUG', 'fatal-warnings,fatal-criticals')
+
```

Comment:
General question... Is there any reason you chose to go with tediously itemizing the individual tests below, instead of looping through a map like you did for the v2 split? Or, did you simply pull in the tests from the former `v1/meson.build` file and strip the "_v1" just to be done with it? https://github.com/fedora-modularity/libmodulemd/blob/ee2b4615e75f388b5efdc29ab4609b2d65007bfe/modulemd/meson.build#L297-L313

Reply:
I just copied them in and stripped the _v1. This branch is not likely to see much activity; I just want it out of the v2 tarball.